amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/ConfigsBuilder.java

```diff
@@ -183,7 +189,7 @@ public void appStreamTracker(Either<MultiStreamTracker, String> appStreamTracker
     }
 
     public void streamTracker(StreamTracker streamTracker) {
-        this.streamTracker = streamTracker;
+        this.streamTracker = StreamARNUtil.trySetStreamARN(streamTracker, this.kinesisRegion);
```

Comment:
Yeah this is what I'm thinking: The StreamTracker can be passed by a customer without a StreamARN set, but even in this situation, we still try to set the StreamARN for the customer internally. So I'm not quite sure how to achieve the Factory mode for this scenario. However, I do understand your point about maintaining the immutability of this object. I think it may only be possible when we expose the StreamARN to the customer